### PR TITLE
NAS-130002 / 24.10 / Inject vendor in main grub.cfg

### PR DESCRIPTION
**Pull request by aiden3c**
This configures the main grub entry to include the vendor name if supplied

### CODE REVIEW

File: src/freenas/usr/local/bin/truenas-grub.py
Comment (creatorcary) [inline src/freenas/usr/local/bin/truenas-grub.py:25]: Using the vendor plugin this becomes
```
with Client() as c:
    vendor = c.call("system.vendor.name") or "TrueNAS Scale"
```
Not sure if this is the preferred way to do it.

Comment (aiden3c) [reply src/freenas/usr/local/bin/truenas-grub.py:25]: I think that since that would require extra imports, it'd be safe to keep it the way it is reading the JSON. We also run truenas-grub from scale-build, I'm not entirely sure what the context looks like there, so just to be safe there too.

Comment (themylogin) [reply src/freenas/usr/local/bin/truenas-grub.py:25]: We can't use middleware client here, middleware is not running during ISO install

**Review by creatorcary — APPROVED** [review]

**Comment by yocalebo** [comment]
please wait for @themylogin final review before merge, thanks!

**Review by themylogin — APPROVED** [review]

**Pull request merged**
